src/client/js/partials/breaches.js

```diff
@@ -104,4 +107,28 @@ function render () {
   renderBreachRows()
 }
 
+async function sendBreachAlertEmail () {
+  try {
+    const res = await fetch('/api/v1/hibp/notify', {
+      method: 'POST',
+      headers: {
+        Authorization: 'Bearer <HIBP_NOTIFY_TOKEN>',
+        'Content-Type': 'application/json',
+        'x-csrf-token': breachesTable.dataset.token
+      },
+      body: JSON.stringify({
+        breachName: 'Adobe',
+        hashPrefix: '365050',
+        hashSuffixes: ['53cbb89874fc738c0512daf12bc4d91765']
+      })
+    })
+
+    if (!res.ok) throw new Error('Bad fetch response')
+
+    console.log('Sent breach alert email')
+  } catch (e) {
+    console.error('Could not send breach alert email:', e)
```

Comment:
Do you intend on landing this test function? It might be better not to catch here, that should make the top-level error handler throw a generic JSON error message.

Reply:
Not exactly in this form, but I’ll intend to have a similar request that is being handled by the controller. I think your comment still applies though!